pkg/expression/helper.go

```diff
@@ -72,11 +72,8 @@ func getTimeCurrentTimeStamp(ctx EvalContext, tp byte, fsp int) (t types.Time, e
 		return value, err
 	}
 	value.SetCoreTime(types.FromGoTime(defaultTime.Truncate(time.Duration(math.Pow10(9-fsp)) * time.Nanosecond)))
-	if tp == mysql.TypeTimestamp || tp == mysql.TypeDatetime || tp == mysql.TypeDate {
-		err = value.ConvertTimeZone(time.Local, ctx.Location())
-		if err != nil {
-			return value, err
-		}
+	if err = value.ConvertTimeZone(defaultTime.Location(), ctx.Location()); err != nil {
+		return value, err
```

Comment:
This is the only file that has some logic change:

1. remove `tp == mysql.TypeTimestamp || tp == mysql.TypeDatetime || tp == mysql.TypeDate` because it seems useless. I check all the places that call `getTimeCurrentTimeStamp`. The tp is always time type. Only converting timezone for time types is also meaningless, so I removed it.

2. Change the first argument of `ConvertTimeZone` from `time.Local` to `defaultTime.Location()` because this the practical meaning for this line. Because `defaultTime` is always in local time zone if it is get from session, it is also right for previous code. But the `StaticEvalContext` returns a current time with the same zone of the context, it's better to make a change.

PTAL @AilinKid

Reply:
here lgtm

Reply:
How about add a `intest.Assert` here?

(It's also fine to not add it, because IMO it's intuitive to call `getTimeCurrentTimeStamp` with time type.